pupil-spa/src/app/services/question/question.service.ts

```diff
@@ -55,9 +57,9 @@ export class QuestionService {
 
     if (currentQuestionNumber === this.getNumberOfQuestions()) {
       // we are already on the last question, there isn't another question number.
+      this.submissionService.submitData().catch(error => new Error(error));
```

Comment:
Putting the submission service here is incongruous, and violates the single responsibility principle.

Reply:
@jon-shipley What's wrong with calling a service from another service?

Reply:
I see what you mean but this block really did nothing before apart from exiting.

Reply:
@GeorgeChatzigiannis  - so a caller calls `getNextQuestionNumber()` an as a side-effect it submits to the backend?  That's not good.

Reply:
@GeorgeChatzigiannis - control flow for the Check itself is managed by the Check Component - the submission must be made from there.